sources/core/Stride.Core.Mathematics/Vector4.cs

```diff
@@ -282,6 +282,39 @@ public float[] ToArray()
             return new float[] { X, Y, Z, W };
         }
 
+        /// <summary>
+        /// Moves the first vector4 to the second one in a straight line.
+        /// </summary>
+        /// <param name="from">The first point.</param>
+        /// <param name="to">The second point.</param>
+        /// <param name="maxTravelDistance">The rate at which the first point is going to move towards the second point.</param>
+        /// <param name="result">When the method completes, contains a version of the Vector that is closer to the second vector.</param>
+        public static void MoveTo(ref Vector4 from, ref Vector4 to, float maxTravelDistance, out Vector4 result)
+        {
+            Vector4 distance;
+            Subtract(ref to, ref from, out distance);
+
+            float length = distance.Length();
+
+            if (maxTravelDistance >= length || length == 0)
+                result = to;
+            else
+                result = new Vector4(from.X + distance.X / length * maxTravelDistance, from.Y + distance.Y / length * maxTravelDistance, from.Z + distance.Z / length * maxTravelDistance, from.W + distance.W / length * maxTravelDistance);
```

Comment:
```
var v = 1f / length * maxTravelDistance;
result = new Vector4(from.X + distance.X * v, from.Y + distance.Y * v, from.Z + distance.Z * v, from.W + distance.W * v);
```
Would be faster, three less divs for one more mult, but it doesn't really matter at this point.

Reply:
Right, its common practice in the stride repo to do that, so indeed recommended.